x-pack/plugins/maps/server/index.ts

```diff
@@ -24,6 +24,35 @@ export const config: PluginConfigDescriptor<MapsXPackConfig> = {
   },
   schema: configSchema,
   deprecations: () => [
+    (
+      completeConfig: Record<string, any>,
+      rootPath: string,
+      addDeprecation: AddConfigDeprecation
+    ) => {
+      if (_.get(completeConfig, 'map.proxyElasticMapsServiceInMaps') === undefined) {
+        return completeConfig;
+      }
+      addDeprecation({
+        documentationUrl:
+          'https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server',
+        message: i18n.translate('xpack.maps.deprecation.proxyEMS.message', {
+          defaultMessage:
+            'map.proxyElasticMapsServiceInMaps is deprecated and will be removed in 8.0.',
+        }),
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.maps.deprecation.proxyEMS.step1', {
+              defaultMessage:
+                'Remove "map.proxyElasticMapsServiceInMaps" in the Kibana config file, CLI flag, or environment variable (in Docker only).',
+            }),
+            i18n.translate('xpack.maps.deprecation.proxyEMS.step2', {
+              defaultMessage: 'Host Elastic Maps Service locally.',
```

Comment:
Can we add hyperlinks in these messages? Would be nice if we can link to the actual EMS-on-prem page.

Reply:
The step text is just treated as a string with no markdown support https://github.com/elastic/kibana/blob/master/x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/steps_modal.tsx#L88

However, the documentationUrl link is visible in several places so there are links to the docs, just not in step 2, but the steps model does display a "View documentation" link.